Makefile

```diff
@@ -0,0 +1,71 @@
+.PHONY: all test gofmt gofmt-fix license-check
+
+# Grab the absolute directory that contains this file.
+ROOT_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+
+
+# These variables can be overridden by setting an environment variable.
+PACKAGES?=pkg/ip subnet network remote
+PACKAGES_EXPANDED=$(PACKAGES:%=github.com/coreos/flannel/%)
+
+default: help
+all: test				    ## Run all the tests
+binary: artifacts/flanneld  ## Create the flanneld binary
+
+flannel_build.created: Dockerfile
+	docker build -t flannel_build .
+	touch flannel_build.created
+
+artifacts/flanneld: flannel_build.created $(shell find . -type f  -name '*.go')
+	mkdir -p artifacts
+	docker run --rm \
+	-v $(ROOT_DIR)/artifacts:/mnt/artifacts \
+	-v $(ROOT_DIR)://go/src/github.com/coreos/flannel:ro \
+	flannel_build bash -c '\
+		go build -o /mnt/artifacts/flanneld -ldflags "-extldflags -static \
+		-X github.com/coreos/flannel/version.Version=$(shell git describe --dirty)" ; \
+		chown -R $(shell id -u):$(shell id -u) /mnt/artifacts'
```

Comment:
It seems we'll need to propagate the makefile go environment to the flannel_build container so it picks up GOARCH, etc.